src/domain/events/EventsPage.tsx

```diff
@@ -160,5 +169,48 @@ const EventsPageMeta: React.FC = () => {
     </Head>
   );
 };
+type UseUpcomingEventsOptions = {
+  variables: UpcomingEventsQueryVariables;
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useUpcomingEvents = ({ variables }: UseUpcomingEventsOptions) => {
+  const {
+    data: eventsData,
+    fetchMore,
+    networkStatus,
+  } = useUpcomingEventsQuery({
+    variables,
+    notifyOnNetworkStatusChange: true,
+  });
+  const events = eventsData?.upcomingEvents?.data;
+  const eventsCount = eventsData?.upcomingEvents?.pageInfo.totalCount;
+  const hasNextPage = eventsData?.upcomingEvents?.pageInfo.hasNextPage;
+  const page = eventsData?.upcomingEvents?.pageInfo.page ?? 0;
+
+  const nextPage = React.useMemo(
+    () => (hasNextPage ? page + 1 : null),
+    [hasNextPage, page]
+  );
```

Comment:
I honestly think that `React.useMemo` is event less perfomant than without.

Reply:
Good point 👍